### DISCUSSION TITLE
Add client to address book/tag via command line

### DISCUSSION THREAD

**Discussion by yaroz** [rustdesk/rustdesk#7866] [General]
I'm using powershell to download and install/configure rustdesk  on several clients.  I'd like to be able to add those clients to a shared address book, and apply a tag to them.  I looked at the source code, and didn't see anything in there to do this.

**Comment by mikekgr** [comment]
Dear @yaroz , if you are ok with that, share your PowerShell script with us. I am looking for a script doing automatic install / config. Thanks for considering.

**Reply by lazee486** [reply]
https://rustdesk.com/docs/en/self-host/client-deployment/  
 the windows one does work, currently using it slightly modified....if you want a simpler way winget does have rusdesk, and then rustdesk --config useconfigkeyhere will configure it. 

the current script assigns a random password, you can use a webhook to telegram or your own server to save it.

**Comment by yaroz** [comment]
> Dear @yaroz , if you are ok with that, share your PowerShell script with us. I am looking for a script doing automatic install / config. Thanks for considering.

Let me do some sanitation on it.

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/commit/e58e75eea9df83b866726cb3ea0f32d3a1a124a1 need to use with server Pro 1.4.2 (will be ready soon)

**Reply by yaroz** [reply]
Just updated to 1.4.2 and downloaded the latest nightly client.  Tried running "rustdesk.exe --address_book_name _IT_" but it did not add it to the address book.  Even tried "rustdesk.exe --assign --token $rustdesktoken --address_book_name _IT_".  Are there updated instructions on how to do this?

disregard for now.. the second should be the correct syntax.  I have 21114 locked down to one address.
